packages/blockchain-wallet-v4/src/redux/webSocket/bch/sagas.js

```diff
@@ -19,9 +20,14 @@ export default ({ api, bchSocket }) => {
 
     switch (message.op) {
       case 'utx':
-        const walletContext = yield select(walletSelectors.getWalletContext)
-        yield put(bchActions.fetchData(walletContext))
         yield put(bchActions.fetchTransactions('', true))
+        const transactions = yield take(bchAT.FETCH_BCH_TRANSACTIONS_SUCCESS)
+        for (let i in transactions.payload.transactions) {
+          const tx = transactions.payload.transactions[i]
+          if (tx.hash === message.x.hash && tx.result > 0) {
+            yield put(A.webSocket.bitcoin.paymentReceived('You\'ve just received a bitcoin payment.'))
+          }
+        }
```

Comment:
v3 does something a bit cheaper and only checks the first tx and I don't remember issues with inconsistency. What do you think?

```
      let tx = wallet.my.wallet.txList.transactions()[0];
      if (tx.result > 0 && tx.txType === 'received') {
        Alerts.displayReceivedBitcoin('JUST_RECEIVED_BITCOIN');
      }```